src/main/java/kelbot/Kelbot.java

```diff
@@ -0,0 +1,115 @@
+                            response = ui.printDone(task);
+                        } else {
+                            Task task = taskList.delete(taskNumber);
+                            response = ui.printDelete(task);
+                        }
+                    } catch (IndexOutOfBoundsException e) {
+                        response = "The list is not that long!";
+                    }
+                } else if (command == Command.FIND) {
+                    try {
+                        if (keyword.equals("")) {
+                            throw new KelbotException("Keyword cannot be empty!");
+                        } else {
+                            TaskList taskListToPrint = new TaskList(taskList.search(keyword));
+                            if (taskListToPrint.toString().equals("")) {
+                                response = "No tasks match your search!";
+                            } else {
+                                response = ui.printRelevantTasks(taskListToPrint);
+                            }
+                        }
+                    } catch (KelbotException e) {
+                        response = e.getMessage();
+                    }
+                } else {
+                    try {
+                        if (taskName == "") {
+                            throw new KelbotException("Task name cannot be empty!");
+                        } else if (command == Command.TODO) {
+                            TodoTask newTodoTask = new TodoTask(taskName);
+                            taskList.add(newTodoTask);
+                            response = ui.printAdd(newTodoTask, taskList.getSize());
+                        } else if (date == null) {
+                            throw new KelbotException("Date cannot be empty!");
+                        } else if (command == Command.DEADLINE) {
+                            DeadlineTask newDeadlineTask = new DeadlineTask(taskName, date);
+                            taskList.add(newDeadlineTask);
+                            response = ui.printAdd(newDeadlineTask, taskList.getSize());
+                        } else {
+                            EventTask newEventTask = new EventTask(taskName, date);
+                            taskList.add(newEventTask);
+                            response = ui.printAdd(newEventTask, taskList.getSize());
+                        }
+                    } catch (KelbotException e) {
+                        response = e.getMessage();
+                    }
+                }
+            } catch (DateTimeParseException e) {
+                response = "Invalid Date!";
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            response = "Which task are you referring to?";
+        } catch (DateTimeParseException e) {
+            response = "Date cannot be empty!";
+        }
+        storage.save(taskList.getTaskList());
+        return response;
+    }
```

Comment:
this deep nested loop can be simplified perhaps? but i actually have something similar myself haha, maybe i should look into it too...